internal/exec/fs.go

```diff
@@ -14,9 +14,9 @@ func CreateDirIfNotExist(dir string) error {
 	return nil
 }
 
-func DeleteFile(location string) error {
-	if err := os.Remove(location); err != nil && !os.IsNotExist(err) {
-		return fmt.Errorf("file %q couldn't be deleted: %w", location, err)
+func DeleteDirectory(location string) error {
+	if err := os.RemoveAll(location); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("%q couldn't be deleted: %w", location, err)
```

Comment:
I know you just renamed the function and instead of files it does directories...

But at that point, the if error can be updated too. There's a difference between `os.Remove` and `os.RemoveAll`, the later says:

> If the path does not exist, RemoveAll returns nil (no error).

So it won't return you a `os.IsNotExists()`, it will just return nil.

It's important to keep that for `os.Remove()` but not for `os.RemoveAll()`.

```suggestion
func DeleteDirectory(location string) error {
	if err := os.RemoveAll(location); err != nil {
		return fmt.Errorf("%q couldn't be deleted: %w", location, err)
```